src/test/java/org/broadinstitute/hellbender/engine/dataflow/transforms/RefBasesFromAPITest.java

```diff
@@ -0,0 +1,110 @@
+import org.broadinstitute.hellbender.utils.dataflow.DataflowUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+public final class RefBasesFromAPITest extends BaseTest {
+
+    @DataProvider(name = "bases")
+    public Object[][] bases(){
+        Object[][] data = new Object[2][];
+        List<Class<?>> classes = Arrays.asList(Read.class, SAMRecord.class);
+        for (int i = 0; i < classes.size(); ++i) {
+            Class<?> c = classes.get(i);
+            DataflowTestData testData = new DataflowTestData(c);
+            List<KV<ReferenceShard, Iterable<GATKRead>>> kvRefShardiReads = testData.getKvRefShardiReads();
+            List<SimpleInterval> intervals = testData.getAllIntervals();
+            List<KV<ReferenceBases, Iterable<GATKRead>>> kvRefBasesiReads = testData.getKvRefBasesiReads();
+            data[i] = new Object[]{kvRefShardiReads, intervals, kvRefBasesiReads};
+        }
+        return data;
+    }
+
+    @Test(dataProvider = "bases")
+    public void refBasesTest(List<KV<ReferenceShard, Iterable<GATKRead>>> kvRefShardiReads,
+                             List<SimpleInterval> intervals, List<KV<ReferenceBases, Iterable<GATKRead>>> kvRefBasesiReads) {
+        Pipeline p = TestPipeline.create();
+        p.getCoderRegistry().registerCoder(ArrayList.class, CollectionCoder.of(new GATKReadCoder()));
+        DataflowUtils.registerGATKCoders(p);
+
+        // Set up the mock for RefAPISource;
+        RefAPISource mockSource = mock(RefAPISource.class, withSettings().serializable());
+        for (SimpleInterval interval : intervals) {
+            when(mockSource.getReferenceBases(any(PipelineOptions.class), any(RefAPIMetadata.class), eq(interval))).thenReturn(FakeReferenceSource.bases(interval));
+        }
+
+        String referenceName = "refName";
+        String refId = "0xbjfjd23f";
+        Map<String, String> referenceNameToIdTable = Maps.newHashMap();
+        referenceNameToIdTable.put(referenceName, refId);
+        RefAPIMetadata refAPIMetadata = new RefAPIMetadata(referenceName, referenceNameToIdTable);
+
+        PCollection<KV<ReferenceShard, Iterable<GATKRead>>> pInput = p.apply(Create.of(kvRefShardiReads));
+
+        PCollection<KV<ReferenceBases, Iterable<GATKRead>>> kvpCollection = RefBasesFromAPI.getBasesForShard(pInput, mockSource, refAPIMetadata);
+        DataflowAssert.that(kvpCollection).containsInAnyOrder(kvRefBasesiReads);
```

Comment:
The problem with this test is that since the reference is all As, we can't tell if the code is grabbing the correct parts of the reference. How about making the fake reference (CAT)* instead?